### PR TITLE
kernel_pca_test.cpp: Add tests

### PR DESCRIPTION
**Pull request by saksham189**
Add tests for kernel_pca command line.

### CODE REVIEW

**Comment by saksham189** [comment]
Thanks for the review. I have made the required changes. Please do see if any other tests are required.

Also, I wanted to point out that when finding the kernel matrix this [line](https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/nystroem_method/nystroem_method_impl.hpp#L89) can cause the entire output to be NaN if a singular value is 0 and then the output dimension is incorrect. Is this supposed to be the expected output or should the NaN values be ignored?

File: src/mlpack/tests/main_tests/kernel_pca_test.cpp [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+  SetInputParam("new_dimensionality", (int) 3);
+  SetInputParam("kernel", (std::string) "badName");
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Make sure if 0 dimensions is specified, we get a dataset with same
```
Comment (ShikharJ) [inline src/mlpack/tests/main_tests/kernel_pca_test.cpp]: I'd prefer if it were something like `Ensure for zero dimensionality, ...`

Comment (rcurtin) [reply src/mlpack/tests/main_tests/kernel_pca_test.cpp]: Done during merge; 1171da89

**Comment by rcurtin** [comment]
Thanks again! :+1:  If you do want to add your name to the list of contributors, you can just open a separate PR, and as before, if you'd like any stickers just send an email.

I've updated the list of completed tests in #1152. :)

File: src/mlpack/tests/main_tests/kernel_pca_test.cpp (hunk 1 above)
Comment (saksham189) [reply src/mlpack/tests/main_tests/kernel_pca_test.cpp]: Thanks. I was traveling for a few days (without laptop) and couldn't make the changes.

**Pull request merged**